masu/external/cost_usage_report_account.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Cost Of Usage Report (CUR) object to be used by Masu Orchestrator/Processor to download CURs."""
+
+
+class CostUsageReportAccount:
```

Comment:
I think I get it, but say once we have a 2nd provider type, that shouldn't matter much here, right? It'll just return an account that will give us enough info to know which provider type to process? Or is this class supposed to be specific to AWS?

Reply:
Thats right, a second provider will work the same way.  The idea of CostUsageReportAccount is to abstract away all of the details of AWS or whatever.  Once we support multiple providers things may have to change slightly to accommodate the other provider but the intention is to be provider agnostic at this point.  

As for the comments I'm using AWS as just an example since that's what the focus is now.